includes/hub/class-nodes.php

```diff
@@ -187,14 +187,31 @@ public static function nodes_dropdown( $current_node = '', $name = 'node_id', $e
 		<?php
 	}
 
+	/**
+	 * Get linking URL.
+	 *
+	 * @param WP_Post $node_post The node post.
+	 */
+	private static function get_site_link_url( $node_post ) {
+		$node_url   = get_post_meta( $node_post->ID, 'node-url', true );
+		$secret_key = get_post_meta( $node_post->ID, 'secret-key', true );
+		return add_query_arg(
+			[
+				'page'       => \Newspack_Network\Node\Settings::PAGE_SLUG,
+				'secret_key' => $secret_key,
```

Comment:
maybe not a huge deal, but it's not best practice to share a secret key like this.

We should create a temporary key, that expires within minutes, that will allow the node to make a request to the Hub to fetch the key

Reply:
I'm not sure this actually would increase security. The secret key would end up in a request URL anyway, and if someone has access to server logs they'd see the URL param from any request. And I don't think the risk warrants the added complexity.

Reply:
Query string [is also encrypted by HTTPS](https://blog.httpwatch.com/2009/02/20/how-secure-are-query-strings-over-https/), but you're right that the secret key would be persisted in browser history. Still a pretty elaborate attack scenario, but I agree it's not ideal.

Reply:
(improved security add in https://github.com/Automattic/newspack-network/pull/64 by @leogermani)